aws/resources.go

```diff
@@ -1328,6 +1342,145 @@ func ec2TransitGatewayMulticastDomain(ctx context.Context, a *aws, resourceType
+	if err != nil {
+		return nil, err
+	}
+
+	resources := make([]provider.Resource, 0)
+
+	for _, i := range transitGatewayPeeringAttachments {
+
+		r, err := initializeResource(a, *i.TransitGatewayAttachmentId, resourceType)
+		if err != nil {
+			return nil, err
+		}
+
+		resources = append(resources, r)
+	}
+
+	return resources, nil
+}
+
+func ec2TransitGatewayPrefixListReferences(ctx context.Context, a *aws, resourceType string, filters *filter.Filter) ([]provider.Resource, error) {
+	routeTablesIDs, err := getTransitGatewayRouteTablesIDs(ctx, a, EC2TransitGatewayRouteTable.String(), filters)
+	if err != nil {
+		return nil, err
+	}
+
+	resources := make([]provider.Resource, 0)
+	for _, tableID := range routeTablesIDs {
+		input := &ec2.GetTransitGatewayPrefixListReferencesInput{
+			TransitGatewayRouteTableId: awsSDK.String(tableID),
+		}
+		prefixListReferences, err := a.awsr.GetTransitGatewayPrefixListReference(ctx, input)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, i := range prefixListReferences {
+			r, err := initializeResource(a, fmt.Sprintf("%s_%s", tableID, *i.PrefixListId), resourceType)
+			if err != nil {
+				return nil, err
+			}
+			resources = append(resources, r)
+		}
+	}
+
+	return resources, nil
+}
+
+func ec2TransitGatewayRoutes(ctx context.Context, a *aws, resourceType string, filters *filter.Filter) ([]provider.Resource, error) {
+	routeTablesIDs, err := getTransitGatewayRouteTablesIDs(ctx, a, EC2TransitGatewayRouteTable.String(), filters)
+	if err != nil {
+		return nil, err
+	}
+
+	resources := make([]provider.Resource, 0)
+	for _, tableID := range routeTablesIDs {
+		input := &ec2.SearchTransitGatewayRoutesInput{
+			Filters: []*ec2.Filter{
+				&ec2.Filter{
+					Name:   awsSDK.String("type"),
+					Values: []*string{awsSDK.String(ec2.TransitGatewayRouteTypeStatic)},
+				},
+			},
+			TransitGatewayRouteTableId: awsSDK.String(tableID),
+		}
+		routes, err := a.awsr.GetTransitGatewayRoutes(ctx, input)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, i := range routes {
+			r, err := initializeResource(a, fmt.Sprintf("%s_%s", tableID, *i.DestinationCidrBlock), resourceType)
+			if err != nil {
+				return nil, err
+			}
+			resources = append(resources, r)
+		}
+	}
+
+	return resources, nil
+}
+
+func ec2TransitGatewayRouteTableAssociations(ctx context.Context, a *aws, resourceType string, filters *filter.Filter) ([]provider.Resource, error) {
+	routeTablesIDs, err := getTransitGatewayRouteTablesIDs(ctx, a, EC2TransitGatewayRouteTable.String(), filters)
+	if err != nil {
+		return nil, err
+	}
+
+	resources := make([]provider.Resource, 0)
+	for _, tableID := range routeTablesIDs {
+		input := &ec2.GetTransitGatewayRouteTableAssociationsInput{
+			TransitGatewayRouteTableId: awsSDK.String(tableID),
+		}
+		associations, err := a.awsr.GetTransitGatewayRouteTableAssociations(ctx, input)
+		if err != nil {
+			return nil, err
+		}
+		// to test with resourceID
```

Comment:
?

Reply:
was a comment that i forgot to remove, doing it now